packages/flame/lib/src/widgets/animation_widget.dart

```diff
@@ -17,24 +18,29 @@ class SpriteAnimationWidget extends StatelessWidget {
   /// Should the animation be playing or not
   final bool playing;
 
-  final Future<SpriteAnimation> Function() _animationFuture;
+  final FutureOr<SpriteAnimation> _animationFuture;
 
   /// A builder function that is called if the loading fails
   final WidgetBuilder? errorBuilder;
 
   /// A builder function that is called while the loading is on the way
   final WidgetBuilder? loadingBuilder;
 
-  SpriteAnimationWidget({
+  const SpriteAnimationWidget({
     required SpriteAnimation animation,
     this.playing = true,
     this.anchor = Anchor.topLeft,
-    this.errorBuilder,
-    this.loadingBuilder,
     Key? key,
-  })  : _animationFuture = (() => Future.value(animation)),
+  })  : _animationFuture = animation,
+        errorBuilder = null,
+        loadingBuilder = null,
         super(key: key);
 
+  /// loads image from asset [path] and renders it as Widget.
+  /// it has [loadingBuilder] while the image from [path] loads image
+  /// To render without loading or gapless playback when the [path] change,
+  /// consider load [SpriteAnimation] from somewhere and directly pass it to
+  /// default constructor
```

Comment:
```suggestion
  /// Loads image from the asset [path] and renders it as a widget.
  ///
  /// It will use the [loadingBuilder] while the image from [path] is loading.
  /// To render without loading, or when you want to have a gapless playback 
  /// when the [path] value changes, consider loading the [SpriteAnimation]
  /// beforehand and direct pass it to the default constructor.
```